server/services/return-of-service.js

```diff
@@ -1,6 +1,111 @@
 /* eslint-disable camelcase */
 const { dbClient, collections } = require('../db');
 const { rosError } = require('../constants');
+const logger = require('../logger.js');
+
+const getParticipantStatuses = async (participantId) => {
+  const statuses = await dbClient.db[collections.PARTICIPANTS_STATUS].find({
+    participant_id: participantId,
+    status: 'hired',
+    current: true,
+  });
+  if (statuses.length === 0) {
+    throw new Error(rosError.participantNotHired);
+  }
+  return statuses;
+};
+
+const getRosParticipant = async (participantId) => {
+  const rosParticipants = await dbClient.db[collections.ROS_STATUS].find({
+    participant_id: participantId,
+    is_current: true,
+  });
+  if (rosParticipants.length === 0) {
+    throw new Error(rosError.participantNotFound);
+  }
+  return rosParticipants[0];
+};
+
+const getDbSiteId = async (site) => {
+  const sites = await dbClient.db[collections.EMPLOYER_SITES].find({
+    'body.siteId': site,
+  });
+  if (sites.length === 0) {
+    throw new Error(rosError.noSiteAttached);
+  }
+  return sites[0].id;
+};
+
+const validateRosUpdateBody = (rosParticipant, updatedSite, updatedDate, updatedStartDate) => {
+  if (!rosParticipant.data) {
+    throw new Error(rosError.fieldNotFound);
+  }
+  if (!updatedSite && !updatedDate && !updatedStartDate) {
+    throw new Error(rosError.noFieldsToUpdate);
+  }
+  if (updatedSite && !rosParticipant.site_id) {
+    throw new Error(rosError.noSiteAttached);
+  }
+  if (updatedDate && !rosParticipant.data.date) {
+    throw new Error(rosError.noDate);
+  }
+  if (updatedStartDate && !rosParticipant.data.startDate) {
+    throw new Error(rosError.noStartDate);
+  }
+};
+
+const formatRosData = async (
```

Comment:
There are still two smells I see here

1. Flag argument in `isMohUpdate` https://martinfowler.com/bliki/FlagArgument.html
2. By the time a service method is called, validation should already have been performed on user input.

In general this is how I would design these methods to reduce conditionals and improve separation of concerns. Don't hesitate to reach out if you want to chat about this

![image](https://user-images.githubusercontent.com/71518072/185676190-dce161af-0479-468d-b042-72a8bdbe315a.png)

Reply:
I do agree that the validation is best performed on the controller / route function.

I'm less against the flag being present in this scenario. Seems like the alternative is two different functions or passing in the user's role, and I don't think I like any of the options more than another. Unless you had a design suggestion for that @arranfw ?

Reply:
My suggestion was to take all of the edit specific functionality and put it in an `updateRosStatus` method